seclang/parser.go

```diff
@@ -77,40 +76,50 @@ func (p *Parser) FromFile(profilePath string) error {
 // or arguments are invalid
 func (p *Parser) FromString(data string) error {
 	scanner := bufio.NewScanner(strings.NewReader(data))
-	var linebuffer = ""
-	pattern := regexp.MustCompile(`\\(\s+)?$`)
+	var linebuffer strings.Builder
 	inQuotes := false
 	for scanner.Scan() {
 		p.currentLine++
 		line := strings.TrimSpace(scanner.Text())
-		if !inQuotes && len(line) > 0 && line[len(line)-1] == '`' {
+		lineLen := len(line)
+		if lineLen == 0 {
+			continue
+		}
+
+		if !inQuotes && line[lineLen-1] == '`' {
 			inQuotes = true
-		} else if inQuotes && len(line) > 0 && line[0] == '`' {
+		} else if inQuotes && line[0] == '`' {
 			inQuotes = false
 		}
+
 		if inQuotes {
-			linebuffer += line + "\n"
-		} else {
-			linebuffer += line
+			linebuffer.WriteString(line)
+			linebuffer.WriteString("\n")
+			continue
+		}
+
+		if line[0] == '#' {
+			continue
 		}
 
 		// Check if line ends with \
-		if !pattern.MatchString(line) && !inQuotes {
-			err := p.evaluate(linebuffer)
+		if line[lineLen-1] == '\\' {
+			linebuffer.WriteString(strings.TrimSuffix(line, "\\"))
```

Comment:
Use slice instead of `TrimSuffix` since we already know the index

Reply:
I was inclined to do that but I wondered if something like

```
SecRule REQUEST_COOKIES|!REQUEST_COOKIES:/__utm/|REQUEST_COOKIES_NAMES|ARGS_NAMES|ARGS|XML:/* "@rx [\r\n]\W*?(?:content-(?:type|length)|set-cookie|location):\s*\w" \
    "id:921120,\\\
    phase:2,\
    block,\
    capture,\
```

is possible where you have more than one `\` at the end of the line cc @fzipi

Reply:
Ah good example. I think if it's supported, having spaces inside is also supposed to be supported, e.g. `\\ \ \`

I don't think either the old or new code handle this. 

Is it possible to define a custom split function for the bufio.Scanner that treats newlines and `\` the same, removing the need to handle it here?

https://pkg.go.dev/bufio#example-Scanner-Custom

Reply:
I am up to work this out if people can provide examples to build the test cases @fzipi @M4tteoP

Reply:
Sorry my idea about customizing the scanner doesn't work for `\` I think - we want it to combine on that, not separate 😣